### ISSUE TITLE
UX: template loading indicator

### ISSUE THREAD

**Issue by jmacura** [OpenOrienteering/mapper#1519]
### Steps to reproduce
1. load large raster template
2. see how the Mapper freezes

### Actual behaviour
Loading large raster templates (especially from HDD drives) takes some time. Until the file is fully loaded (it can take up to minutes), the program freezes and the user can only hope that it will unfreeze in the near future.

### Expected behaviour
Minimal fix: Show some kind of loader to indicate, that something is really happening in behind and that it's not a permanent freeze. This loader can be either simple (e.g. spinner) or (better) showing % of loaded data.
Cool fix: Load the file asynchronously, show % of loaded data in status bar and let the user interact with the program meanwhile.

### Configuration
Mapper Version: 0.9.1, latest master (20191221.17)
Operating System: Win 10 Pro v1903

**Comment by dg0yt** [comment]
With the debug versions I use most of the time, loading is even slower. So this topic ranks high on my todo list, and I already did some work in that direction. The final solution is to do the loading in the background. But this also means another internal template state ("Waiting for load"), so it is not entirely trivial.

**Comment by lpechacek** [comment]
This report mostly duplicates issue 1243, if I'm not mistaken. @jmacura, are you fine with merging your ticket into the older one?

For the record, the work done on the CoVe concurrent processing routines may perhaps be reused for the core Mapper code.

**Comment by lpechacek** [comment]
Oh, only now I've noticed dg0yt's comment, therefore scratch my one. The remark about duplication is still valid, I believe.

**Comment by jmacura** [comment]
refs #1243

**Comment by jmacura** [comment]
@lpechacek I agree those two issues are about the same problem, only appearing in a different context.
I let on your workflow and customs if you prefer to merge it into one issue or not. I am fine with either option.

**Comment by dg0yt** [comment]
I think this issue here is not a duplicate but just one sub-issue of #1243.

**Comment by lpechacek** [comment]
When there is a volunteer to do the work with good quality, all discussions about work organization are moot. Apologies for the noise. I've messed with this issue too much.

**Comment by dg0yt** [comment]
@jmacura Is your main concern actually the loading of a single (new) template, or rather the loading of a map with multiple templates?

**Comment by jmacura** [comment]
@dg0yt I think both of these are equally annoying. Improving either of these will be highly appreciated. Yet the second (loading of a map with many templates) can be mitigated by turning off the templates before save, so I would put a slightly higher priority to the second – loading of a single large template.
